### PR TITLE
Add llvm.*max and llvm.*min intrinsics translation

### PR DESCRIPTION
**Pull request by MochalovaAn**
Add llvm.smax, llvm.smin , llvm.umax, llvm.umin intrinsic translation and add test for that

Signed-off-by: amochalo <anastasiya.mochalova@intel.com>

### CODE REVIEW

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -2320,6 +2324,38 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
                                   transValue(II->getArgOperand(1), BB)};
     return BM->addExtInst(STy, BM->getExtInstSetId(SPIRVEIS_OpenCL), ExtOp, Ops,
                           BB);
+  }
+    // trata
```
Comment (Fznamznon) [inline lib/SPIRV/SPIRVWriter.cpp:2328]: lol :)
```suggestion
```

File: lib/SPIRV/SPIRVWriter.cpp [hunk 2]
```diff
@@ -2320,6 +2324,38 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
...
+    case (Intrinsic::smax):
+      OC = OpSGreaterThan;
+      break;
+    case (Intrinsic::umin):
+      OC = OpULessThan;
+      break;
+    default:
+      OC = OpUGreaterThan;
+      break;
+    }
```
Comment (Fznamznon) [inline lib/SPIRV/SPIRVWriter.cpp:2351]: This switch on Intrinsic ID inside the switch on Intrinsic ID seems weird. I would suggest to add something like `getBuiltinIdForIntrinsic` function or extend it so it handles not only instructions from OCL extended set or handle each intrinsic separately.

**Review by AlexeySachkov — APPROVED** [review]

**Review by Fznamznon — APPROVED** [review]

File: lib/SPIRV/SPIRVWriter.cpp [hunk 3]
```diff
@@ -2321,6 +2325,35 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
...
+        BM->addCmpInst(OC, transType(BoolTy), FirstArgVal, SecondArgVal, BB);
+    return BM->addSelectInst(Cmp, FirstArgVal, SecondArgVal, BB);
+  }
+  case Intrinsic::smin:
+  case Intrinsic::smax: {
+    Type *BoolTy = IntegerType::getInt1Ty(M->getContext());
+    SPIRVValue *FirstArgVal = transValue(II->getArgOperand(0), BB);
+    SPIRVValue *SecondArgVal = transValue(II->getArgOperand(1), BB);
+
+    Op OC = (II->getIntrinsicID() == Intrinsic::smin) ? OpSLessThan
```
Comment (mlychkov) [inline lib/SPIRV/SPIRVWriter.cpp:2349]: The only difference in code between these 2 blocks is op code.
Processing of these 4 intrinsics might be done in one block to reduce duplication of code.

**Review by mlychkov — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**